### PR TITLE
fix: Finish TTFD when binding transaction to scope

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Finish the TTFD span before creating a new transaction and binding it to the scope in SentryUIViewControllerPerformanceTracker when the user doesn't call reportFullyDisplayed.

## :bulb: Motivation and Context

Fixes GH-4513

## :green_heart: How did you test it?
Unit tests and simulator.

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by brustolin — APPROVED** [review]
LGTM

File: Sources/Sentry/SentryTimeToDisplayTracker.m [hunk 1]
```diff
@@ -137,6 +137,22 @@ - (void)reportFullyDisplayed
...
+    if (self.initialDisplaySpan.isFinished == NO) {
+        [self.initialDisplaySpan finish];
+    }
+
+    if (self.fullDisplaySpan.isFinished == NO) {
+        SENTRY_LOG_WARN(
+            @"You didn't call SentrySDK.reportFullyDisplayed(). Finishing full display span with "
+            @"status deadline exceeded.");
+
+        [self.fullDisplaySpan finishWithStatus:kSentrySpanStatusDeadlineExceeded];
```
Comment (brustolin) [inline Sources/Sentry/SentryTimeToDisplayTracker.m:151]: m: I just dont know if this is a good status for this case.

Comment (philipphofmann) [reply Sources/Sentry/SentryTimeToDisplayTracker.m:151]: I also thought about that. It's the same as when you forget to call reportFullyDisplayed, and the 30s timeout hits. I wanted to keep it consistend.

**Comment by brustolin** [comment]
BTW, I just realize a problem that goes like this:

- ViewControllerA opens, start a http request request, user clicks a button and open ViewControllerB.
- ViewControllerB starts a http request
- ViewControllerA http request ends and `SentrySDK.reportFullyDisplayed` is called


This will finish TTFD for ViewControllerB

**Comment by philipphofmann** [comment]
> BTW, I just realized a problem that goes like this:
> 
> * ViewControllerA opens, starts a HTTP request, user clicks a button and opens ViewControllerB.
> * ViewControllerB starts an HTTP request.
> * ViewControllerA’s HTTP request ends and `SentrySDK.reportFullyDisplayed` is called.
> 
> This will finish TTFD for ViewControllerB.

Do you think this PR adds this problem or that the problem still exists and needs to be addressed @brustolin?

**Comment by brustolin** [comment]
>Do you think this PR adds this problem or that the problem still exists and needs to be addressed @brustolin?

I think this PR adds this problem. Of course, by fixing another problem, we need to choose.

> Why do you think the ViewControllerA’s HTTP request ending does call SentrySDK.reportFullyDisplayed? Do you mean the user calls it? 

The user would call it in the code. 

> Before this PR calling SentrySDK.reportFullyDisplayed when being on ViewControllerB ends finishes TTFD of ViewControllerB. The same applies to this PR, but the difference now is that the TTFD of UIViewControllerA doesn't timeout any more.

🤔 I need to check If I can reproduce what Im claiming.

**Comment by philipphofmann** [comment]
I'm not sure if this PR adds the problem. Yes, please try if you can reproduce it, ideally with a test.

File: Tests/SentryTests/Integrations/Performance/UIViewController/SentryUIViewControllerPerformanceTrackerTests.swift [hunk 2]
```diff
@@ -766,6 +766,137 @@ class SentryUIViewControllerPerformanceTrackerTests: XCTestCase {
...
+        }
+        
+        XCTAssertEqual(secondTracer?.traceId, SentrySDK.span?.traceId)
+        
+        XCTAssertTrue(firstFullDisplaySpan.isFinished)
+        XCTAssertEqual(.deadlineExceeded, firstFullDisplaySpan.status)
+        
+        let secondFullDisplaySpan = try XCTUnwrap(secondTracer?.children.first { $0.operation == "ui.load.full_display" })
+        
+        XCTAssertFalse(secondFullDisplaySpan.isFinished)
```
Comment (brustolin) [inline Tests/SentryTests/Integrations/Performance/UIViewController/SentryUIViewControllerPerformanceTrackerTests.swift:814]: h: I dont get this test, how could this `true` even before this PR?

**Comment by brustolin** [comment]
@philipphofmann I wrote the test to reproduce what Im talking about in [here](https://github.com/getsentry/sentry-cocoa/pull/4531)

File: Tests/SentryTests/Integrations/Performance/UIViewController/SentryUIViewControllerPerformanceTrackerTests.swift (hunk 2 above)
Comment (brustolin) [reply Tests/SentryTests/Integrations/Performance/UIViewController/SentryUIViewControllerPerformanceTrackerTests.swift:812]: > the secondFullDisplaySpan didn't even exist

Ok, this is the problem. Got it.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**